### PR TITLE
⚠️ Remove v1alpha5 API Version

### PR DESCRIPTION
**Pull request by kashifest**
Remove the v1alpha5 version across the CAPM3 codebase.
Related to #971

### CODE REVIEW

**Comment by mboukhalfa** [comment]
seems not triggered :no_good:

**Comment by kashifest** [comment]
/hold

There are more places in e2e where we need to drop alpha api

**Comment by adilGhaffarDev** [comment]
If we want to add the new clusterctl upgrade tests(`v0.3=>v1.5=>current`) that CAPI has added, we should keep these APIs in our code. 
ref: https://github.com/kubernetes-sigs/cluster-api/pull/10147

**Comment by kashifest** [comment]
> If we want to add the new clusterctl upgrade tests(`v0.3=>v1.5=>current`) that CAPI has added, we should keep these APIs in our code. ref: [kubernetes-sigs/cluster-api#10147](https://github.com/kubernetes-sigs/cluster-api/pull/10147)

ok interesting, I was following this, https://github.com/kubernetes-sigs/cluster-api/pull/9939. Are you sure that we are going to test (`v0.3=>v1.5=>current`)  we dont have 0.3 APIs either, perhaps you meant (`v0.4=>v1.5=>current`) for CAPI and (`v0.5=>v1.5=>current`) for us? But then, I am also thinking didnt we decide to drop upgrade from alpha versions ? 

The reason I started to work on this now is when I bump to CAPI `v1.7.0-beta.0` , I am getting the following error 

` sigs.k8s.io/cluster-api/api/v1alpha4: package sigs.k8s.io/cluster-api/api/v1alpha4 provided by sigs.k8s.io/cluster-api at latest version v1.6.3 but not at required version v1.7.0-beta.0`

**Comment by adilGhaffarDev** [comment]
> ok interesting, I was following this, [kubernetes-sigs/cluster-api#9939](https://github.com/kubernetes-sigs/cluster-api/pull/9939). Are you sure that we are going to test (`v0.3=>v1.5=>current`) we dont have 0.3 APIs either, perhaps you meant (`v0.4=>v1.5=>current`) for CAPI and (`v0.5=>v1.5=>current`) for us? But then, I am also thinking didnt we decide to drop upgrade from alpha versions ?

In CAPI they moved alpha APIs to an internal folder, they removed it from the API folder that is on root. 
Also, CAPI didn't ask providers to add these tests, they only asked to uplift, so we are fine. These tests are running in CAPI already and the edge case where the upgrade fails is already covered there we don't need to test it on our side. We can remove alpha APIs for now. Sorry for the confusion.

**Comment by kashifest** [comment]
/override test-ubuntu-e2e-feature-main 
ubuntu feature test is failing for unrrelated reason, centos has passed

**Comment by kashifest** [comment]
Passed tests:

- https://jenkins.nordix.org/job/metal3_capm3_main_integration_test_ubuntu/1031/
- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_integration_test_ubuntu/23/
- https://jenkins.nordix.org/job/metal3_capm3_from-release-1-6_main_e2e_clusterctl_upgrade_test_ubuntu/32/
- https://jenkins.nordix.org/job/metal3_capm3_from-release-1-5_main_e2e_clusterctl_upgrade_test_ubuntu/9/
- https://jenkins.nordix.org/job/metal3_capm3_from-release-1-4_main_e2e_clusterctl_upgrade_test_ubuntu/20/
- https://jenkins.nordix.org/job/metal3_capm3_from-release-1-3_main_e2e_clusterctl_upgrade_test_ubuntu/5/
- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_integration_test_centos/718/
- https://jenkins.nordix.org/job/metal3_capm3_main_e2e_feature_test_centos/102/

File: test/e2e/upgrade_clusterctl_test.go [hunk 1]
```diff
@@ -274,8 +274,8 @@ func preInitFunc(clusterProxy framework.ClusterProxy) {
 	})
 
 	// Export capi/capm3 versions
-	os.Setenv("CAPI_VERSION", "v1alpha4")
-	os.Setenv("CAPM3_VERSION", "v1alpha5")
+	os.Setenv("CAPI_VERSION", "v1beta1")
+	os.Setenv("CAPM3_VERSION", "v1beta1")
```
[the remark this answers is not in the document]
Comment (mboukhalfa) [reply test/e2e/upgrade_clusterctl_test.go:278]: also overriding the format is in line 283

[the remark this answers is not in the document]
Comment (kashifest) [reply test/e2e/upgrade_clusterctl_test.go:278]: I dont think I understood your 2nd comment , if these vars are not needed, we need to do cleanup in a followup.

File: test/e2e/config/e2e_conf.yaml [hunk 2]
```diff
@@ -39,7 +39,7 @@ providers:
 - name: kubeadm
   type: BootstrapProvider
   versions:
-  - name: ${CAPI_FROM_RELEASE} # latest published release in the v1alpha4 series; this is used for v1alpha4 --> v1beta1 clusterctl upgrades test only.
+  - name: ${CAPI_FROM_RELEASE}
```
Comment (furkatgofurov7) [inline test/e2e/config/e2e_conf.yaml:42]: I can't directly comment on the line, but what I am not understanding is why this is so much complicated rather then just using hardcoded values for CAPI_FROM and CAP_TO releases. That is not my main question though, but it is this line:

```
files:
    - sourcePath: "../data/shared/${CAPI_FROM_RELEASE:0:4}/metadata.yaml"
```

how it used to work AND will work as is? you have `v0.4`, `v1.2`, `v1.3`, `v1.4`, `v1.5`, `v1.6` folders under `data/shared` folder

Comment (kashifest) [reply test/e2e/config/e2e_conf.yaml:42]: I agree this is hard to follow and confusing as well. We have another PR where we are hardcoding these values for simplifications, https://github.com/metal3-io/cluster-api-provider-metal3/pull/1465/, as for how it works, @mboukhalfa can clarify more, we are exporting `CAPI_TO_RELEASE` inherits its value from `CAPIRELEASE` which is set in dev-env. I think I have to also do a cleanup of the line you pointed out.

Comment (mboukhalfa) [reply test/e2e/config/e2e_conf.yaml:42]: that line is extract the the minor release and removing the patch chars to set the correct path based on the release exported in dev-env

File: test/e2e/upgrade_clusterctl_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mboukhalfa) [reply test/e2e/upgrade_clusterctl_test.go:278]: ok
